src/installer/__main__.py

```diff
@@ -0,0 +1,210 @@
+            target_path = pathlib.Path(self.scheme_dict[scheme], record.path)
+            if sys.version_info < (3, 9):
+                compileall.compile_file(target_path, optimize=level)
+            else:
+                compileall.compile_file(
+                    target_path,
+                    optimize=level,
+                    stripdir=str(self.destdir),
+                )
+
+    def finalize_installation(
+        self,
+        scheme: Scheme,
+        record_file_path: str,
+        records: Iterable[Tuple[Scheme, RecordEntry]],
+    ) -> None:
+        record_list = list(records)
+        super().finalize_installation(scheme, record_file_path, record_list)
+        for scheme, record in record_list:
+            self._compile_record(scheme, record)
+
+
+def main_parser() -> argparse.ArgumentParser:
+    """Construct the main parser."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("wheel", type=str, help="wheel file to install")
+    parser.add_argument(
+        "--destdir",
+        "-d",
+        metavar="/",
+        type=str,
+        default="/",
+        help="destination directory (prefix to prepend to each file)",
+    )
+    parser.add_argument(
+        "--optimize",
+        "-o",
+        nargs="*",
+        metavar="level",
+        type=int,
+        default=(0, 1),
+        help="generate bytecode for the specified optimization level(s) (default=0, 1)",
+    )
+    parser.add_argument(
+        "--skip-dependency-check",
+        "-s",
+        action="store_true",
+        help="don't check if the wheel dependencies are met",
+    )
+    return parser
+
+
+def get_scheme_dict(distribution_name: str) -> Dict[str, str]:
+    """Calculate the scheme disctionary for the current Python environment."""
+    scheme_dict = sysconfig.get_paths()
+
+    # calculate 'headers' path, sysconfig does not have an equivalent
+    # see https://bugs.python.org/issue44445
+    dist_dict = {
+        "name": distribution_name,
+    }
+    distribution = distutils.dist.Distribution(dist_dict)
+    install_cmd = distribution.get_command_obj("install")
+    assert install_cmd
+    install_cmd.finalize_options()
+    # install_cmd.install_headers is not type hinted
+    scheme_dict["headers"] = install_cmd.install_headers  # type: ignore
+
+    return scheme_dict
+
+
+def check_python_version(metadata: Message) -> None:
+    """Check if the project support the current interpreter."""
+    try:
+        import packaging.specifiers
+    except ImportError:
+        warnings.warn(
+            "'packaging' module not available, "
+            "skipping python version compatibility check"
+        )
+        return
+
+    requirement = metadata["Requires-Python"]
+    if not requirement:
+        return
+
+    versions = requirement.split(",")
+    for version in versions:
+        specifier = packaging.specifiers.Specifier(version)
+        if platform.python_version() not in specifier:
```

Comment:
Maybe use `sysconfig.get_python_version()` instead since that should generally work better for cross compilation scenarios I think.

Reply:
Good point, but unfortunately `sysconfig.get_python_version` does not contain the minor version.

Reply:
Docs indicate [it does](https://docs.python.org/3/library/sysconfig.html#sysconfig.get_python_version):
> Return the `MAJOR.MINOR` Python version number as a string.

Reply:
Sorry! I meant the patch version.